protocols/kad/src/behaviour.rs

```diff
@@ -732,6 +732,26 @@ where
     /// The result of the query is delivered in a
     /// [`Event::OutboundQueryProgressed{QueryResult::GetClosestPeers}`].
     pub fn get_closest_peers<K>(&mut self, key: K) -> QueryId
+    where
+        K: Into<kbucket::Key<K>> + Into<Vec<u8>> + Clone,
+    {
+        self.get_closest_peers_inner(key, None)
+    }
+
+    /// Initiates an iterative query for the closest peers to the given key.
+    ///
+    /// The result of the query is delivered in a
+    /// [`Event::OutboundQueryProgressed{QueryResult::GetClosestPeers}`].
+    ///
+    /// The expected responding peers is specified by `num_results`
+    pub fn get_closest_peers_num_results<K>(&mut self, key: K, num_results: NonZeroUsize) -> QueryId
```

Comment:
I wonder whether it would be better to have a public function `get_closest_peers_with_config` accepting a `Config` including `num_results` instead of `get_closest_peers_num_results`. This would allow to keep a lean interface in the future if more parameters are added to the `get_closest_peers` function?

Reply:
So far, the `Config` itself is not exposed outside.
Also, the population of the `Config` might requires access to the `current` config, as the calling place may not have the knowledge of other config variants.

Hence, in this PR, it's being chosen to just pass `num_results` to minimize the change.